### DISCUSSION TITLE
ContextFlyout is invalid

### DISCUSSION THREAD

**Discussion by Maklith** [AvaloniaUI/Avalonia#15624] [Q&A]
There's a weird problem with ContextFlyout partially taking effect
This ContextFlyout is only available on the controls2:NodeInput template
But look at other controls where ContextFlyout is present but has no effect
https://github.com/MakesYT/Kitopia/blob/Avalonia/KitopiaAvalonia/Windows/TaskEditor.axaml#L1200
![51f6b52280962d8957c31faddd2507fb](https://github.com/AvaloniaUI/Avalonia/assets/42534870/560b6579-53c6-4714-b231-61005801e2b4)
![0bdeeafca29cf5ac43f5657ff95cdf4a](https://github.com/AvaloniaUI/Avalonia/assets/42534870/8a43ed35-c150-48d5-9092-bbc24b536ee3)
Normally it would look something like this
![image](https://github.com/AvaloniaUI/Avalonia/assets/42534870/321f9468-a48f-4859-9567-15dd77098120)
But in fact it displays the ContextFlyout of the parent control
![image](https://github.com/AvaloniaUI/Avalonia/assets/42534870/90e2fea3-aa85-4a8d-a7e4-152bb6a14226)

**Comment by timunie** [comment]
Does the control have a solid background? Transparent is okay but not null.

**Reply by Maklith** [reply]
Yes,
![image](https://github.com/AvaloniaUI/Avalonia/assets/42534870/b4f34bbe-7988-4556-b97d-3cf2de07741c)

**Reply by timunie** [reply]
I see null in DevTools here. file a very minmal sample App

**Reply by Maklith** [reply]
![image](https://github.com/AvaloniaUI/Avalonia/assets/42534870/509546c1-5301-4ea8-91a8-ec06143eecd1)
This program is open source

**Reply by timunie** [reply]
> This program is open source

We are not going to debug your App, also not if it is open source.

**Reply by Maklith** [reply]
I created a minimal example
https://github.com/MakesYT/ContextFlyout

**Reply by timunie** [reply]
For me it looks all as expected. Node has the Menu set, so all children show it if they haven't set it to anything else. That is exactly what it does for me.

**Reply by Maklith** [reply]
The child controls are actually set, but the settings are invalid due to a capture error from the control library.
I have solved the problem, thank you for your help
